Makefile

```diff
@@ -2,6 +2,7 @@ CFLAGS = -ggdb
 CXXFLAGS = ${CFLAGS} -fsanitize=address
 LDFLAGS = -fsanitize-memory-track-origins -fsanitize-memory-use-after-dtor
 
+BASE_DIR = $(realpath $(dir $(realpath $(firstword $(MAKEFILE_LIST)))))
```

Comment:
Why realpath twice?

Reply:
`dir` leaves the trailing slash and `realpath` removes it. Would work without it but I just think it's cleaner that way.

Reply:
Hm, the inner call should still be redundant.

Reply:
The inner `realpath` call? `$(firstword $(MAKEFILE_LIST))` that resolves to the location of the `Makefile` which I think is relative and in the simplest case just `Makefile` so we need to turn it into a path before feeding to `dir`.